### PR TITLE
support json columns like chat_history

### PR DESCRIPTION
**Pull request by RichardLi1437**
1. use multiple rows instead of first row to infer schema
2. support json fields like chat_history
3. add test cases

### CODE REVIEW

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py [hunk 1]
```diff
@@ -178,18 +179,32 @@ def read_data(row):
...
+    data_as_df = spark.createDataFrame(infer_pdf)
+    # data_as_df.show()  
+    # data_as_df.printSchema()
 
     # The temporary workaround to remove the chat_history column if it exists.
     # We are removing the column because the pyspark DF is unable to parse it.
     # This version of the MDC is applied only to GSQ.
-    if MDC_CHAT_HISTORY_COLUMN in data_as_df.columns:
-        data_as_df = data_as_df.drop(col(MDC_CHAT_HISTORY_COLUMN))
+    # if MDC_CHAT_HISTORY_COLUMN in data_as_df.columns:
```
Comment (yunjie-hub) [inline assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:197]: We still need to drop it now

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py [hunk 2]
```diff
@@ -178,18 +179,32 @@ def read_data(row):
         return data_url
         # TODO: Move this to tracking stream if both data and dataref are NULL
 
-    # Output MLTable
+    def row_to_pdf(row) -> pd.DataFrame:
+        return pd.read_json(read_data(row))
+
     data_columns = _get_data_columns(df)
-    first_data_row = df.select(data_columns).rdd.map(lambda x: x).first()
+    data_rows = df.select(data_columns).rdd.take(SCHEMA_INFER_ROW_COUNT)  # TODO: make it an argument user can define
```
Comment (yunjie-hub) [inline assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:186]: Can we do a sample 10 rows?

Comment (RichardLi1437) [reply assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:186]: the sample() only take fraction, no row_count, :(

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 from fsspec import AbstractFileSystem
 from azureml.fsspec import AzureMachineLearningFileSystem
 from datetime import datetime
-from pyspark.sql.functions import col, lit
```
Comment (yunjie-hub) [inline assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:18]: Please upgrade the component verison

Comment (RichardLi1437) [reply assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py:18]: upgraded

File: assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py (hunk 1 above)
Comment (RichardLi1437) [reply assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py]: dropped

**Review by yunjie-hub — APPROVED** [review]

**Review by yunjie-hub — APPROVED** [review]

**Pull request merged**